### ISSUE TITLE
GitHub CLA Bot Fix

### ISSUE THREAD

**Issue by natectang** [sourcegraph/sourcegraph#25318]
After speaking with Dan Adler on the issue with the GitHub CLA Bot, just coming here to open up an issue to get some more context and track progress on getting the CLA Bot fixed. Thanks for your time

**Comment by bobheadxi** [comment]
Some context-building:

https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/configure/sgbot/sgbot.Deployment.yaml#L41
https://github.com/sourcegraph/infrastructure/pull/2576

**Comment by bobheadxi** [comment]
@tammy-zhu @srj427

**Comment by bobheadxi** [comment]
I think we might want to dump sgbot, and switch to something a bit more hands-off: https://github.com/cla-assistant/cla-assistant

**Comment by bobheadxi** [comment]
tl;dr of reasoning: from https://github.com/sourcegraph/analytics/issues/117 and skimming over the code, what I'm pretty sure happened here is that the CLA spreadsheet was previously public and editable by anyone:

- There is nothing indicating credentials are provided to `sgbot` https://sourcegraph.com/github.com/sourcegraph/deploy-sourcegraph-dot-com@release/-/blob/configure/sgbot/sgbot.Deployment.yaml?L29-42
- Fetching the sheet uses a plain `GET` request: https://sourcegraph.com/github.com/sourcegraph/infrastructure/-/blob/docker-images/sgbot/vendor/github.com/sourcegraph/maintainerbot/tasks/tasks.go?L361-380

I imagine this stopped working when access was restricted to the doc (currently requires sign-in) for obvious reasons (we ask for addresses, phone numbers, etc.)

Since a cursory search didn't reveal anything about being able to use simple auth headers to download spreadsheets (most google APIs require auth keys) so I'm assuming the only way to do this would be to actually use the Google sheets API. This isn't strictly _hard_ but will require some effort:

- what auth would look like: https://github.com/sourcegraph/sourcegraph/blob/main/internal/cmd/resources-report/README.md#google-cloud-platform
- what updating sheets looks like: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/internal/cmd/resources-report/sheets.go

I wrote the above code so have some familiarity with doing this. If none of the off-the-shelf solutions look good, I think I can write the related code + spin up the bot again etc in ~2-3 days but IMO an off-the-shelf solution would be better in the long run.

**Comment by tammy-zhu** [comment]
@bobheadxi I'm supportive of us using an off-the-shelf solution. If further legal review of solutions would help you make a decision, let me know. If the team knows which solution it wants, then go for it - no need for further legal review. cc @dadlerj @natectang @srj427

**Comment by natectang** [comment]
@bobheadxi Hey Robert, I'd love to touch base with you when you get back next week. I ended up choosing the Colin Eberhardt CLA Bot: https://colineberhardt.github.io/cla-bot and while it was working temporarily earlier this morning, I'm not too sure what happened but it stopped checking CLAs an hour in. I created two test repos to apply the bot, and within each config file, I tried using a URL that returns a JSON array of CLA signatures (using github handles) for the bot to check for, but by the time I got to this point, the CLA bot had stopped working. I can provide more context when we meet next week.

<img width="924" alt="Screen Shot 2021-10-01 at 10 29 31 AM" src="https://user-images.githubusercontent.com/15318653/135697509-5fa478d0-a699-4720-8de8-21850c5d2df0.png">

**Comment by bobheadxi** [comment]
Where are the test repositories? Can I look at the configuration as well, and get a link to the example you have in your screenshot?
